File: go/person/person.go
package person

import "fmt"

type Person struct {
	Name     string
	LastName string
	Age      int
}

func (p Person) GetFullName() string {
	return fmt.Sprintf("%s %s", p.Name, p.LastName)
}

type PersonInterface interface {
	GetFullName() string
}

File: elixir/lib/Person/person.ex
defmodule Person do
  defstruct firstName: "", lastName: "", age: 0

  defimpl PersonInterface, for: Person do
    def firstName(), do: Person.firstName()
    def lastName(), do: Person.lastName()
    def age(), do: Person.age()
  end

  def getFullName(%Person{firstName: firstName, lastName: lastName}) do
    "#{firstName} #{lastName}"
  end
end
